litmus-portal/graphql-server/pkg/database/mongodb/operations.go

```diff
@@ -44,32 +44,34 @@ func UpdateCluster(query bson.D, update bson.D) error {
 	return nil
 }
 
-func UpsertWorkflowRun(workflow_id string, wfRun WorkflowRun) error {
+func UpsertWorkflowRun(workflow_id string, wfRun WorkflowRun) (int, error) {
 	opts := options.Update().SetUpsert(true)
 	ctx, _ := context.WithTimeout(backgroundContext, 10*time.Second)
 
 	count, err := workflowCollection.CountDocuments(ctx, bson.M{"workflow_id": workflow_id, "workflow_runs.workflow_run_id": wfRun.WorkflowRunID})
 	if err != nil {
-		return err
+		return 0, err
 	}
 
+	updateCount := 1
```

Comment:
In which case, update count will be 0?

Reply:
If the `completed` param is true and not false